optimum/intel/openvino/quantization.py

```diff
@@ -330,25 +329,86 @@ def _quantize_torchmodel(
             model_type=model_type,
         )
 
-        if self.task == "text-generation":
-            onnx_config = onnx_config_class(self.model.config, use_past=self.model.config.use_cache)
+        if task == "text-generation":
+            onnx_config = onnx_config_class(model.config, use_past=model.config.use_cache)
         else:
-            onnx_config = onnx_config_class(self.model.config)
+            onnx_config = onnx_config_class(model.config)
 
-        compressed_model.eval()
-        num_parameters = compressed_model.num_parameters()
-        save_as_external_data = use_external_data_format(num_parameters) or quantization_config.save_onnx_model
-        f = io.BytesIO() if not save_as_external_data else save_directory / ONNX_WEIGHTS_NAME
+        onnx_path = save_directory / ONNX_WEIGHTS_NAME
 
-        # Export the compressed model to the ONNX format
+        # Export the model to the ONNX format
         opset = min(onnx_config.DEFAULT_ONNX_OPSET, MAX_ONNX_OPSET)
-        opset = opset if not quantization_config.save_onnx_model else max(opset, MIN_ONNX_QDQ_OPSET)
-        _onnx_export_nncf_model(compressed_model, onnx_config, f, opset)
+        opset = max(opset, MIN_ONNX_QDQ_OPSET)
+        export(
+            model=compressed_model,
+            config=onnx_config,
+            opset=opset,
+            output=onnx_path,
+        )
 
         # Load and save the compressed model
-        model = core.read_model(f) if save_as_external_data else core.read_model(f.getvalue(), b"")
+        model = core.read_model(onnx_path)
         self._save_pretrained(model, output_path)
         quantization_config.save_pretrained(save_directory)
+        if not quantization_config.save_onnx_model:
+            os.remove(onnx_path)
```

Comment:
It seems this removes the ONNX model, but not the model.onnx_data

Reply:
export(...) already does most of the work, I need only read from the .onnx and remove it if it is not needed. 
There reason, why I save to the file, comes from LLMs, this helps to avoid some issues. Anyway, next month we are planning to replace it with direct export from pytorch to OV model which speed ups the whole export process significantly. The work on the OV side is almost done.

Reply:
That's great news! But I didn't mean that the ONNX saving is problematic, it's just that for large models this also creates a model.onnx_data which is not removed. I think we could just add:
```python
try:
    os.remove(f"{onnx_path}_data")
except FileNotFoundError:
    pass
```